### DISCUSSION TITLE
Environment Secrets - Where is this located and how can I edit or delete this?

### DISCUSSION THREAD

**Discussion by inbytesdevteam** [vercel/community#538] [Help] [answered]
Where is the 'Environment Secrets' stored? I can create environment secret directly in the UI when I am adding an env, but I can't find where I can edit or delete that secret value. 

Also with the new update, it's written that "It's no longer necessary to create Secrets separately. Just add the value above." So can I assume that secrets are deprecated. 

And I had this issue in the env that value should only be string if you select the "Secret" option. I have added a string and it still sees it not a string so I create a env secret for it or set it to plain text in order for it to work.

Thanks.

**Comment by swarnava** [comment, marked as answer]
Hi there,

The Environment variable settings page of your project should also let you edit the existing Environment Variable:

<img width="901" alt="image" src="https://user-images.githubusercontent.com/1162991/161540455-08519c86-1536-47f9-b5a9-f9cbd9b468fd.png">

If you are deploying using `.env` file, a new deployment should recalibrate based on your latest changes. Learn more here: https://vercel.com/docs/concepts/projects/environment-variables

**Reply by inbytesdevteam** [reply]
Hi @swarnava thanks for the reply. 

There is this SECRET that you can create in Vercel thru the environment setting but I think it was deprecated. However I still have access to it. I want to update or delete it but I don't know where because I don't know where to find it in the UI or something.

<img width="837" alt="Screen Shot 2022-04-05 at 9 34 11 AM" src="https://user-images.githubusercontent.com/70461143/161662244-2c782450-94ed-4dab-b5fd-07dd1e21e8d2.png">

<img width="447" alt="Screen Shot 2022-04-05 at 9 34 35 AM" src="https://user-images.githubusercontent.com/70461143/161662251-c5ec6bb2-de9b-4147-af50-107ad62845b3.png">

**Reply by Pieparker** [reply]
Hey there, [this article](https://vercel.com/support/articles/how-do-i-migrate-away-from-vercel-json-env-and-build-env) should tell you everything you need to know! In short, they may be in a `vercel.json` file.

**Reply by inbytesdevteam** [reply]
Hi @Pieparker, thank you for this. :) I have been using the environment variables UI in configuring my secrets so no secrets are placed in vercel.json. I'm having a hard time finding where it is stored in Vercel so I can update or delete it. I have [read](https://vercel.com/changelog/environments-variables-per-git-branch) that there is no need to create Secrets anymore. So I was planning to remove all the Secrets save in my vercel account and update it as suggested.

I've noticed that it is stored somewhere:

For example: If I create the same Secret name, it will just tell me that a Secret has already exist even thought I have deleted the env var. So it must be save somewhere in the system.

**Reply by Pieparker** [reply]
In which case, you can [use the CLI](https://vercel.com/docs/cli#commands/secrets) to identify secrets!

**Reply by inbytesdevteam** [reply]
Thank you!!! 😊
